apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/DataCompactionSparkApp.java

```diff
@@ -58,6 +65,23 @@ protected void runInner(Operations ops) {
           fileGroupRewriteResult.rewrittenDataFilesCount(),
           fileGroupRewriteResult.rewrittenBytesCount());
     }
+    // Add compaction timestamp to table properties.
+    CompactionEvent event =
+        CompactionEvent.builder()
+            .compactionTimestamp(System.currentTimeMillis())
+            .addedDataFilesCount(result.addedDataFilesCount())
+            .rewrittenDataFilesCount(result.rewrittenDataFilesCount())
+            .rewrittenBytesCount(result.rewrittenBytesCount())
+            .build();
+    Gson gson = new GsonBuilder().create();
+    Type type = new TypeToken<CompactionEvent>() {}.getType();
+    ops.spark()
+        .sql(
+            String.format(
+                "ALTER TABLE %s SET TBLPROPERTIES ('%s' = '%s')",
+                ops.getTable(fqtn),
+                DATA_LAYOUT_COMPACTION_PROPERTY_KEY,
+                StringEscapeUtils.escapeJava(gson.toJson(event, type))));
```

Comment:
We should move it to https://github.com/linkedin/openhouse/blob/main/libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/persistence/StrategiesDaoTableProps.java

Reply:
Hmm, but it's not related to the strategy, right? It's a compaction event. Should I push this functionality into a separate util?